### ISSUE TITLE
Crash in git_describe_commit due to attempt to free uninitialized pointer

### ISSUE THREAD

**Issue by apnadkarni** [libgit2/libgit2#6272]
### Reproduction steps

Console session:

```
D:\Temp>git clone \src\libgit2\tests\resources\testrepo.git lg2-test
Cloning into 'lg2-test'...
done.

D:\Temp>cd lg2-test

D:\Temp\lg2-test>..\lg2-142-release\lg2 tag
annotated_tag_to_blob
e90810b
hard_tag
point_to_blob
taggerless
test
hard_tag

D:\Temp\lg2-test>..\lg2-142-release\lg2 describe point_to_blob
```

### Expected behavior

An error message should be printed because `point_to_blob` does not refer to a committish.

### Actual behavior

lg2 program crashes.

### Version of libgit2 (release number or SHA1)

1.4.2

### Operating system(s) tested

Windows

**Comment by apnadkarni** [comment]
Pull request in #6273

**Comment by ethomson** [comment]
Fixed via #6273
